integrations/operator/controllers/resources/role_controller.go

```diff
@@ -49,7 +49,7 @@ var TeleportRoleGVKV5 = schema.GroupVersionKind{
 type RoleReconciler struct {
 	kclient.Client
 	Scheme                 *runtime.Scheme
-	TeleportClientAccessor sidecar.ClientAccessor
+	TeleportClientAccessor syncclient.ClientAccessor
```

Comment:
this should be named as client or teleport.ClientAccessor.
It doesn't do any sync at all and just replaces the client when credentials are reloaded and no other reconciler uses the client

Reply:
I renamed it to `protectedclient.Accessor` as this is a regular Teleport client with a protection mechanism so it is not closed while we use it